in_toto/models/layout.py

```diff
@@ -248,6 +269,38 @@ def __init__(self, **kwargs):
 
     self.validate()
 
+  def __repr__(self):
+    """
+    <Purpose>
+      Represents a readable format of Step Class fields
+    <Returns>
+      Step Class fields, i.e. name (threshold), pubkeys 
+      expected_command, expected_materials, expected_products
+    """
+    heading = "\n    step name: {} (threshold: {})".format(self.name, self.threshold)
+    pubkeys = "  pubkeys: {}".format("\n  ".join(self.pubkeys))
+    if not self.expected_command:
+      expected_command = "  expected cmd: None"
+    else:
+      expected_command = "  expected cmd: {}".format(" ".join(self.expected_command))
+
+    expected_materials = "  expected input (materials):\n\t"
+    if not self.expected_materials:
+      expected_materials += "   None"
+    else:
+      for value in self.expected_materials:
+        expected_materials += "   {}\n\t".format(" ".join(value))
```

Comment:
Might be worth writing a little `_display_artifact_rule()` helper function since you have copied this block 4 times.

Reply:
I created a `check_string()` function in util.py to address some issues of copy/pasting the same code over and over again.